crates/wasmi/src/table.rs

```diff
@@ -28,11 +28,11 @@ pub enum TableError {
     /// Occurs when growing a table out of its set bounds.
     GrowOutOfBounds {
         /// The maximum allowed table size.
-        maximum: usize,
+        maximum: u32,
         /// The current table size before the growth operation.
-        current: usize,
+        current: u32,
         /// The amount of requested invalid growth.
-        grow_by: usize,
+        grow_by: u32,
```

Comment:
Since Wasm is a 32-bit platform we technically have to allow for allocations of up to 2^32 bytes in size at least on 64-bit platforms (and above). These sizes are not possible to represent using a `u32` value since in fact the limit is going to be `u32::MAX + 1` on 64-bit platforms and `i32::MAX as u32 + 1` on 32-bit platforms.

Reply:
Sure? I agree these types should be usize.
But other TableType must use `u32`

Reply:
No, as I have already stated the maximum table size exceeds `u32::MAX` and therefore `TableType` needs to be built on `usize`.

Reply:
I only know memory64, Do wasm also support table64?

Reply:
A new idxtype can be either i32 or i64

idxtype ::= i32 | i64
The [limits](https://webassembly.github.io/spec/core/syntax/types.html#syntax-limits) structure is parameterised by index syntax

limits_iv ::= {min iv, max iv?} The parameter is omitted where it is immaterial.
The [table type](https://webassembly.github.io/spec/core/syntax/types.html#table-types) continues to use i32 indices

tabletype ::= limits_i32 elemtype
The [memory type](https://webassembly.github.io/spec/core/syntax/types.html#memory-types) structure is extended to have an index type

memtype ::= idxtype limits_iv (iff idxtype = type(iv))